### PR TITLE
Backport perl version testing macros; regenerate to 5.33.0

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

**Comment by khwilliamson** [comment]
Actually, I see I rebased wrong.  Though a small matter, the regenerate commit should be swapped with the backport commit.  The regeneration was done afterwards

File: HACKERS [hunk 1]
```diff
@@ -204,20 +204,22 @@ Perl public API.
...
+Version checking used to be tricky to get correct (besides being buggy in some
+perl versions).
+C<ivers()> is used in the C<=tests> section to overcome this. and constructs
 like the following in the C language sections.
 
-  #if { VERSION < 5.9.3 }
+  #if PERL_VERSION_LT(5,9,3)
+  #if PERL_VERSION_GT(5,9,3)
+  #if PERL_VERSION_LE(5,9,3)
+  #if PERL_VERSION_GE(5,9,3)
```
Comment (atoomic) [inline HACKERS:215]: maybe mention the `EQ` and `NE` flavors?

File: parts/inc/inctools [hunk 2]
```diff
@@ -28,7 +28,7 @@ sub parse_version
 
   my($r,$v,$s);
 
-  if (   ($r, $v, $s) = $ver =~ /^(5)(\d{3})(\d{3})$/ # 5029010, from the file
+  if (   ($r, $v, $s) = $ver =~ /^(\d)(\d{3})(\d{3})$/ # 5029010, from the file
```
Comment (atoomic) [inline parts/inc/inctools:31]: maybe use `/a` as `\d` could be too wise and means more than `[0-9]`

File: parts/inc/version [hunk 3]
```diff
@@ -41,11 +42,12 @@ PERL_PATCHLEVEL_H_IMPLICIT
...
-#endif /* PERL_REVISION != 5 */
+#define D_PPP_RVS_TO_BCD(r,v,s) ((D_PPP_DEC2BCD(r)<<24)|(D_PPP_DEC2BCD(v)<<12)|D_PPP_DEC2BCD(s))
+#define PERL_BCDVERSION D_PPP_RVS_TO_BCD(PERL_REVISION, PERL_VERSION, PERL_SUBVERSION)
+
+__UNDEFINED__ PERL_VERSION_EQ(r,v,s) (PERL_BCDVERSION == D_PPP_RVS_TO_BCD(r,v,s))
+__UNDEFINED__ PERL_VERSION_NE(r,v,s) (! PERL_VERSION_EQ(r,v,s))
+__UNDEFINED__ PERL_VERSION_LT(r,v,s) (PERL_BCDVERSION < D_PPP_RVS_TO_BCD(r,v,s))
+__UNDEFINED__ PERL_VERSION_LE(r,v,s) (PERL_BCDVERSION <= D_PPP_RVS_TO_BCD(r,v,s))
+__UNDEFINED__ PERL_VERSION_GT(r,v,s) (! PERL_VERSION_LE(r,v,s))
+__UNDEFINED__ PERL_VERSION_GE(r,v,s) (! PERL_VERSION_LT(r,v,s))
```
Comment (atoomic) [inline parts/inc/version:53]: this is cool using the `!` versions :-)

**Pull request merged**